spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -0,0 +1,864 @@
+            .setBody(objNode.toString().getBytes())
+            .execute();
+
+        assertThat(adminClient.owners().getOwner(scaOwner.getKey()))
+            .returns(Owners.ENTITLEMENT_ACCESS_MODE, OwnerDTO::getContentAccessMode)
+            .returns(Owners.ENTITLEMENT_ACCESS_MODE, OwnerDTO::getContentAccessModeList);
+    }
+
+    @Test
+    public void shouldProduceAContentAccessCertificateForTheConsumerOnRegistration() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        String ownerKey = owner.getKey();
+
+        ProductDTO prod = adminClient.ownerProducts().createProductByOwner(ownerKey, Products.random());
+        ContentDTO content = adminClient.ownerContent().createContent(ownerKey, Content.random());
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+
+        ConsumerApi consumerApi = new ConsumerApi(adminClient.getApiClient());
+        Object export = consumerApi.exportCertificates(consumer.getUuid(), null);
+        List<JsonNode> certs = CertificateUtil
+            .extractEntitlementCertificatesFromPayload(export, ApiClient.MAPPER);
+        assertThat(certs).singleElement();
+        Map<String, List<String>> prodIdToContentIds = CertificateUtil.toProductContentIdMap(certs.get(0));
+        assertThat(prodIdToContentIds)
+            .hasSize(1)
+            .extractingByKey("content_access", as(collection(String.class)))
+            .hasSize(1)
+            .containsExactly(content.getId());
+
+        JsonNode certContent = certs.get(0).get("products").get(0).get("content").get(0);
+        assertEquals(content.getType(), certContent.get("type").asText());
+        assertEquals(content.getName(), certContent.get("name").asText());
+        assertEquals(content.getLabel(), certContent.get("label").asText());
+        assertEquals(content.getVendor(), certContent.get("vendor").asText());
+        assertEquals(0, certContent.get("arches").size());
+        verifyCertContentPath(ownerKey, content.getContentUrl(), null, certContent.get("path").asText());
+
+        List<String> payloadCerts = extractCertsFromPayload(export);
+        assertThat(payloadCerts).singleElement();
+        byte[] value = CertificateUtil
+            .compressedContentExtensionValueFromCert(payloadCerts.get(0), "1.3.6.1.4.1.2312.9.7");
+        X509HuffmanDecodeUtil decode = new X509HuffmanDecodeUtil();
+        List<String> urls = decode.hydrateContentPackage(value);
+        assertThat(urls).contains(CandlepinMode.isStandalone() ? "/" + ownerKey : "/sca/" + ownerKey);
+
+        String extVal = CertificateUtil
+            .standardExtensionValueFromCert(payloadCerts.get(0), "1.3.6.1.4.1.2312.9.8");
```

Comment:
It might be worth extracting the magic strings like `1.3.6.1.4.1.2312.9.8` into named constants to make it obvious what we are extracting.

Reply:
Added ENT_DATA_OID and ENT_TYPE_OID as named constants. If I'm understanding the OIDs correctly, I think the names make sense.